js/lib/AlphabeticIndex.js

```diff
@@ -60,159 +60,291 @@ var Collator = require("./Collator.js");
  * asynchronously. If this option is given as "false", then the "onLoad"
  * callback must be given, as the instance returned from this constructor will
  * not be usable for a while.
  *
- * <li><i>loadParams</i> - an object containing parameters to pass to the 
+ * <li><i>loadParams</i> - an object containing parameters to pass to the
  * loader callback function when locale data is missing. The parameters are not
- * interpretted or modified in any way. They are simply passed along. The object 
+ * interpretted or modified in any way. They are simply passed along. The object
  * may contain any property/value pairs as long as the calling code is in
  * agreement with the loader callback function as to what those parameters mean.
  * </ul>
- * 
+ *
  * @constructor
  * @param {Object} options options to the parser
  */
 var AlphabeticIndex = function (options) {
 	this.sync = true;
 	this.loadParams = {};
-	
+	this.caseSensitive = false;
+	this.accentSensitive = false;
+	this.overflowLabel = "#";
+	this.underflowLabel = "*";
+	this.style = "standard";	
+	this.index = [];
+
 	if (options) {
 		if (options.locale) {
 			this.locale = (typeof(options.locale) === 'string') ? new Locale(options.locale) : options.locale;
 		}
 		
+		if (typeof(options.style) !== 'undefined') {
+			this.style = options.style;
+		}
+
+		if (typeof(options.overflowLabel) !== 'undefined') {
+			this.overflowLabel = options.overflowLabel;
+		}
+
+		if (typeof(options.underflowLabel) !== 'undefined') {
+			this.underflowLabel = options.underflowLabel;
+		}
+		
 		if (typeof(options.sync) !== 'undefined') {
 			this.sync = (options.sync == true);
 		}
-		
 		if (options.loadParams) {
 			this.loadParams = options.loadParams;
 		}
 	}
 
 	this.locale = this.locale || new Locale();
-
-	//console.log("implemented in pure JS");
-	if (!Collator.cache) {
-		Collator.cache = {};
-	}
-
+	
 	Utils.loadData({
-		object: Collator, 
-		locale: this.locale, 
+		object: "AlphabeticIndex",
+		locale: this.locale,
 		name: "collation.json",
-		sync: sync,
-		loadParams: loadParams, 
+		sync: this.sync,
+		loadParams: this.loadParams, 
+
 		callback: ilib.bind(this, function (collation) {
 			if (!collation) {
 				collation = ilib.data.collation;
 				var spec = this.locale.getSpec().replace(/-/g, '_');
-				Collator.cache[spec] = collation;
+				ilib.data.cache.Collator[spec] = collation;
 			}
+			this.collation = collation;
 			this._init(collation);
-			new LocaleInfo(this.locale, {
-				sync: sync,
-				loadParams: loadParams,
-				onLoad: ilib.bind(this, function(li) {
-					this.li = li;
-					if (this.ignorePunctuation) {
-		    			isPunct._init(sync, loadParams, ilib.bind(this, function() {
-							if (options && typeof(options.onLoad) === 'function') {
-								options.onLoad(this);
-							}
-		    			}));
-	    			} else {
-						if (options && typeof(options.onLoad) === 'function') {
-							options.onLoad(this);
-						}
-	    			}
-	    		})
-			});
+
```

Comment:
We use CType below, but we don't initialize it here in the constructor to allow it to load its json files.